Format source code

bjoernhaeuser opened the pull request:
I want to make the source code a bit nicer to read and mor consistent. I was not able to find a general jenkins plugins style guide so I just applied the general one from intellij.

@samrocketman what do you think?

bjoernhaeuser commented:
If we want to go down this road I would apply the changes to all code.

samrocketman commented:
I'm not against changing code style.  However, if we're really going forward with this we're missing two things:

1. Documentation:  like `STYLE_GUIDE.md` and a reference to the style guiding in `CONTRIBUTING.md`
2. Automated code style validation which runs with the tests.  People suck at strict code style so we really need a pass/fail test for code style.

Otherwise, I'm not sure it's worth opening this can of worms to make tons of style changes.  The docs serve as a reference when people contest the style and the automated validation serve to force people to correct style before their change can be merged.

bjoernhaeuser commented:
@samrocketman agreed!

What I did:

* Added checkstyle maven plugin
* Added `checkstyle.xml` and adapted a bit to the current project
* Work on the violations bring the amount of violations to zero
* Added a `STYLE_GUIDE.md` which is now rather short on context

samrocketman commented:
Something to be aware of: merging this will cause merge conflicts on a lot of open PRs.  We have to be willing to help (even resolve) with some of these conflicts.

bjoernhaeuser commented on src/main/java/org/jenkinsci/plugins/ghprb/GhprbTriggerBackwardsCompatible.java:
```diff
@@ -1,10 +1,10 @@
+// CHECKSTYLE:OFF
```
It has a ton of violations which are hard to fix. This class looks like something we should remove soonish (next major version maybe).

bjoernhaeuser commented:
I agree with everything you said.

samrocketman commented:
Great job.  I know this kind of stuff is a lot of work.  A huge attention to detail.
